src/VisualStudio/Core/Impl/ProjectSystem/CPS/TempPECompiler.cs

```diff
@@ -49,16 +49,31 @@ public async Task<bool> CompileAsync(IWorkspaceProjectContext context, string ou
                 cancellationToken.ThrowIfCancellationRequested();
             }
 
-            var options = project.LanguageServices.GetRequiredService<ICompilationFactoryService>().GetDefaultCompilationOptions()
+            // We need to inherit most of the projects options, mainly for VB (RootNamespace, GlobalImports etc.), but we need to override about some specific things surrounding the output
+            var options = project.CompilationOptions
                     // copied from the old TempPE compiler used by legacy, for parity.
                     // See: https://github.com/dotnet/roslyn/blob/fab7134296816fc80019c60b0f5bef7400cf23ea/src/VisualStudio/CSharp/Impl/ProjectSystemShim/TempPECompilerService.cs#L58
                     .WithAssemblyIdentityComparer(DesktopAssemblyIdentityComparer.Default)
                     .WithSourceReferenceResolver(SourceFileResolver.Default)
                     .WithXmlReferenceResolver(XmlFileResolver.Default)
-                    // we always want to produce a DLL
-                    .WithOutputKind(OutputKind.DynamicallyLinkedLibrary);
+                    // We always want to produce a debug, AnyCPU DLL
+                    .WithOutputKind(OutputKind.DynamicallyLinkedLibrary)
+                    .WithPlatform(Platform.AnyCpu)
+                    .WithOptimizationLevel(OptimizationLevel.Debug)
+                    // Turn off any warnings as errors just in case
+                    .WithGeneralDiagnosticOption(ReportDiagnostic.Suppress)
+                    .WithReportSuppressedDiagnostics(false)
+                    .WithSpecificDiagnosticOptions(null)
+                    // Turn off any signing and strong naming
+                    .WithDelaySign(false)
+                    .WithCryptoKeyFile(null)
+                    .WithPublicSign(false)
+                    .WithStrongNameProvider(null);
 
-            project = project.WithCompilationOptions(options);
+            project = project
+                .WithCompilationOptions(options)
+                // AssemblyName should be set to the filename of the output file because multiple TempPE DLLs can be created for the same project
```

Comment:
I admit your "because..." isn't very clear here. You're already writing to your own files, but you need the assembly name to match that too?

Reply:
Given a project with an assembly name of `CoreWindowsFormsApp1` for example, we might need to produce two TempPE DLLS, `Resource1.Designer.cs.dll` and `DataSet1.Designer.cs.dll`. Now that we're inheriting the AssemblyName from the project this becomes a problem, as both DLLs have the same assembly name.

Reply:
Is this because you're loading these into an app domain or something and the actual embedded assembly name matters?

Reply:
Not us, but VSTypeResolutionService loads them into an app domain yes. Also when a user creates a new data source in the Data Sources window a `.datasource` file is created in the project, which is an XML file that has the assembly name in it for the specific TempPE DLL that the data source comes from.